util/reggen/uvm_reg.sv.tpl

```diff
@@ -110,11 +111,20 @@ package ${block.name}_ral_pkg;
 %>\
     % if tag[0] == "excl":
       csr_excl.add_excl(${f.name}.get_full_name(), ${tag[2]}, ${tag[1]});
-      % endif
+    % endif
     % endfor
   % endif
 % endfor
-
+% if reg_shadowed:
+  % for r_tag in r.tags:
+<% tag = r_tag.split(":") %>\
```

Comment:
It looks like the tag always needs to be specified in the hjson but we don't check if it's not specified for a shadow register. Should there be a check for that?

Ideally, the tag would have to be provided just for `hwext` registers. For normal shadow registers placed in `u_reg` the regtool should be able to derive `shadow_reg_path` automatically right?

Reply:
That is a good point, thank you for your suggestion. Yes I will add a default option for non-hwext registers so they do not need to add a tag